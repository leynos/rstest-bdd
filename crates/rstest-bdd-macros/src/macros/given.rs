//! Implementation of the `#[given]` macro.

use proc_macro::TokenStream;

/// Macro for defining a Given step that registers with the step inventory.
pub(crate) fn given(attr: TokenStream, item: TokenStream) -> TokenStream {
    super::step_attr(attr, item, crate::StepKeyword::Given)
}
