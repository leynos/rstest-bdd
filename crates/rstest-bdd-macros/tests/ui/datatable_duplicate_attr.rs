//! Compile-fail fixture: duplicate `#[datatable]` attribute on a single parameter is rejected.

use rstest_bdd_macros::given;

#[given("duplicate attribute")]
fn step(#[datatable] #[datatable] table: Vec<Vec<String>>) {}

fn main() {}
