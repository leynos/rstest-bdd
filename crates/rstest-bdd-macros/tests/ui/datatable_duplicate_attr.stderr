error: invalid step function signature: duplicate `#[datatable]` attribute. Hint: use `#[given] fn name(ctx: &StepContext, ... )` and valid fixtures.
 --> tests/ui/datatable_duplicate_attr.rs:6:35
  |
6 | fn step(#[datatable] #[datatable] table: Vec<Vec<String>>) {}
  |                                   ^^^^^

error: duplicate `#[datatable]` attribute
 --> tests/ui/datatable_duplicate_attr.rs:6:35
  |
6 | fn step(#[datatable] #[datatable] table: Vec<Vec<String>>) {}
  |                                   ^^^^^
